File: src/lira/linguistics/data/linguistic_unit.py
"""Base type for every artefact in the Word/Clause/Sentence/Paragraph/
Subject/UserPrompt tree (Layer Summary: Linguistics Layer)."""

from dataclasses import dataclass
from typing import Optional


@dataclass
class LinguisticUnit:
    text: str
    system_property: Optional["LinguisticSystemProperty"] = None

File: prototype/src/lira/linguistics/data/linguistic_unit.ts
import type { LinguisticSystemProperty } from "./system_property";

/** Base type for every artefact in the Word/Clause/Sentence/Paragraph/
 * Heading/Document/UserPrompt tree (Layer Summary: Linguistics Layer). Ported
 * from linguistics/data/linguistic_unit.py. `systemProperty` is now
 * fully typed against the real `LinguisticSystemProperty` (the
 * Linguistics Service is ported -- see linguistics/role/) -- Vocabulary's
 * `Word` still only ever reads/writes it as an opaque field, same as
 * before. */
export interface LinguisticUnit {
  text: string;
  systemProperty?: LinguisticSystemProperty;
}
